### PR TITLE
feat: rvc import gui (gui only)

### PR DESCRIPTION
**Pull request by iMoonDay**
### Description/描述

实现Import GUI，但未完善具体Import功能

添加2个按键对应Import GUI (R+I)和Export GUI (R+E)

### Checklist/清单

- [ ] The issues this PR addresses are related to each other/此 PR 解决的 issue 是相关的
- [ ] My changes are all related to the related issue above/我的修改都与上方的相关 issue 有关
- [x] My changes generate no new warnings/我的修改没有产生新的警告
- [x] My code builds and runs on my machine/我的代码在我的电脑上可以编译和运行
- [ ] I documented my code/我已经为我的代码添加了注释/文档

### Screenshots
image

### CODE REVIEW

File: src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt [hunk 1]
```diff
@@ -1,4 +1,171 @@
...
+                TODO()
+            }
+        },
+        Other("Other") {
+            override fun addChildren(screen: SelectionImportScreen, rootComponent: FlowLayout) {
+                File("schematics").mkdirs()
+                File("schematics").listFiles()!!.asSequence()
+                    .filterNot { it.isDirectory || it.name.endsWith(".litematic") } // ignore litematica
+                    .filter { (it.extension in setOf("schematic", "schem")) }
+                    .forEach { rootComponent.child(screen.FileLine(it)) }
```
Comment (Qubik65536) [inline src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt]: Why here's also Litematica related operations?

Comment (iMoonDay) [reply src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt:160]: 这部分是 @zly2006 修改的，并非我的源代码，我也不清楚

Comment (iMoonDay) [reply src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt]: This part was modified by @zly2006 and is not my source code, so I don’t know

Comment (Qubik65536) [reply src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt]: > This part was modified by @zly2006 and is not my source code, so I don’t know

https://github.com/zly2006/reden-is-what-we-made/pull/86/commits/21620cc7e17e83899ee2e7fc9895318d3835f636#diff-cf4acc03943058b07d58a08062a131e3775a55be73eead4566f96f823fe6de86R195-R206

But it is from your commit.

Comment (iMoonDay) [reply src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt]: I once submitted a PR, but it was lost. This PR is the final code after all previous modifications.

Comment (Qubik65536) [reply src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt]: > I once submitted a PR, but it was lost. This PR is the final code after all previous modifications.

Ok then

Comment (iMoonDay) [reply src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt]: This part was modified in this branch: [fix: gui bugs, changed hotkey config order](https://github.com/zly2006/reden-is-what-we-made/pull/86/commits/52a96965bc138eaff17941c53289f94fb2bd458b)

**Review by Qubik65536 — APPROVED** [review]
LGTM

File: src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt (hunk 1 above)
Comment (Qubik65536) [reply src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt]: OK I get it then

Comment (zly2006) [reply src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt]: The category "Other" refers to some schematic files that are not litematica's, usually created by worldedit or some other tools. They are in a similar format to structure block.

**Comment by zly2006** [comment]
The last commit:
1. added `name` field.
2. Improved structure block file display, e.g. `namespace:a` now is shown as `namespace:a` not `a` (namespace `minecraft` is still ignored by default)
3. **removed useless tooltips**, do we still need that?

File: src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt [hunk 2]
```diff
@@ -1,4 +1,172 @@
...
+            gap(5)
+            alignment(HorizontalAlignment.LEFT, VerticalAlignment.CENTER)
+            child(importButton)
+        })
+    }
+
+    inner class FileLine(
+        val file: File,
+        val name: String,
+        vararg tooltips: Text,
```
Comment (zly2006) [inline src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt]: tooltips已经不再需要

File: src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt [hunk 3]
```diff
@@ -1,4 +1,172 @@
...
+        }
+    }
+
+    fun Long.formatToDate(): String {
+        val localDateTime = LocalDateTime.ofInstant(Instant.ofEpochMilli(this), ZoneId.systemDefault())
+        val formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")
+        return localDateTime.format(formatter)
+    }
+
+    enum class Type(val displayName: String) {
```
Comment (zly2006) [inline src/main/java/com/github/zly2006/reden/rvc/gui/SelectionImportScreen.kt]: 显示相关的文本都要用Text表示，再仔细检查一下你的代码please

**Review by zly2006 — CHANGES_REQUESTED** [review]
再考虑到一个全新的需求： #87，你必须添加另一个分类实现读取.rvcarchive文件，此文件也在schematics目录下，简单实现一下吧

**Review by zly2006 — APPROVED** [review]

**Pull request merged**